internal/users/manager.go

```diff
@@ -217,6 +288,12 @@ func (m *Manager) UserByName(username string) (UserEntry, error) {
 // UserByID returns the user information for the given user ID.
 func (m *Manager) UserByID(uid uint32) (UserEntry, error) {
 	usr, err := m.cache.UserByID(uid)
+	if errors.Is(err, cache.NoDataFoundError{}) {
+		// Check if the user is a temporary user.
+		if user, ok := m.temporaryUsers[uid]; ok {
```

Comment:
Shouldn't we lock the mutex to access this?

Reply:
done